fuel-p2p/src/peer_info.rs

```diff
@@ -0,0 +1,384 @@
+#[derive(Debug)]
+pub struct PeerInfo {
+    peer_addresses: HashSet<Multiaddr>,
+    client_version: Option<String>,
+    connected_point: ConnectedPoint,
+    latest_ping: Option<Duration>,
+}
+
+impl PeerInfo {
+    pub fn new(conencted_point: ConnectedPoint) -> Self {
+        Self {
+            peer_addresses: HashSet::default(),
+            client_version: None,
+            connected_point: conencted_point,
+            latest_ping: None,
+        }
+    }
+}
+
+// `Behaviour` that holds info about peers
+pub struct PeerInfoBehaviour {
+    ping: Ping,
+    identify: Identify,
+    peers: HashMap<PeerId, PeerInfo>,
+}
+
+impl PeerInfoBehaviour {
+    pub fn new(local_public_key: PublicKey) -> Self {
+        let identify = {
+            let cfg = IdentifyConfig::new("/fuel/1.0".to_string(), local_public_key);
+            Identify::new(cfg)
+        };
+
+        Self {
+            ping: Ping::new(PingConfig::new()),
+            identify,
+            peers: HashMap::default(),
+        }
+    }
+
+    pub fn get_peer_info(&self, peer_id: &PeerId) -> Option<&PeerInfo> {
+        self.peers.get(peer_id)
+    }
+
+    /// Insert peer addresses to a connected Node
+    pub fn insert_peer_addresses(&mut self, peer_id: &PeerId, addresses: Vec<Multiaddr>) {
+        match self.peers.get_mut(peer_id) {
+            Some(peer_info) => {
+                for address in addresses {
+                    peer_info.peer_addresses.insert(address);
+                }
+            }
+            _ => {
+                debug!(target: "fuel-libp2p", "Peer with PeerId: {:?} is not among the connected peers", peer_id)
+            }
+        }
+    }
+
+    /// Store newly connected peer
+    fn insert_peer(&mut self, peer_id: &PeerId, connected_point: ConnectedPoint) {
+        match self.peers.get_mut(peer_id) {
+            Some(peer_info) => {
+                peer_info.connected_point = connected_point;
+            }
+            _ => {
+                self.peers.insert(*peer_id, PeerInfo::new(connected_point));
+            }
+        }
+    }
+
+    /// Insert client version to a connected Node
+    fn insert_client_version(&mut self, peer_id: &PeerId, client_version: String) {
+        match self.peers.get_mut(peer_id) {
+            Some(peer_info) => {
+                peer_info.client_version = Some(client_version);
+            }
+            _ => {
+                debug!(target: "fuel-libp2p", "Peer with PeerId: {:?} is not among the connected peers", peer_id)
+            }
+        }
+    }
+
+    /// Insert latest ping to a connected Node
+    fn insert_latest_ping(&mut self, peer_id: &PeerId, duration: Duration) {
+        match self.peers.get_mut(peer_id) {
+            Some(peer_info) => {
+                peer_info.latest_ping = Some(duration);
+            }
+            _ => {
+                debug!(target: "fuel-libp2p", "Peer with PeerId: {:?} is not among the connected peers", peer_id)
+            }
+        }
```

Comment:
if let might be cleaner than match here

```suggestion
        if let Some(peer_info) = self.peers.get_mut(peer_id) {
            peer_info.latest_ping = Some(duration);
         } else {
             debug!(target: "fuel-libp2p", "Peer with PeerId: {:?} is not among the connected peers", peer_id)
        }
```